docs/running_as_a_service.md

```diff
@@ -0,0 +1,150 @@
+for this document) on Linux through systemd, Mac through launchd or Windows
+through Windows services. It is recommended to run the ConCom agent as a service
+for tasks such as:
+* Checking machine specific status such as SSH root login being enabled, etc.
+  The specifics can be configured as per the documentation
+  [here](configuration.md).
+* Running specific checks related to internal network security.
+* Running as a server to do checks against external machines or APIs, etc.
+* Running in a container inside a Kubernetes cluster to check for cluster
+  compliance.
+
+The one-off process is more useful for things like:
+* Running in CI/CD pipeline to check code compliance of code in that pipeline.
+* Running in cron jobs to check the compliance at regular intervals.
+* Running in a container to check the compliance of the container image.
+
+This document details how to run the ConCom agent as a service, server or daemon
+on Linux, Mac and Windows. For details of how to run as a one-off process see
+[here](running_as_a_process.md).
+
+You should read the steps for the particular OS you are interested in. The steps
+should be independent of CPU architecture though you should ensure you download
+the agent for your chosen architecture.
+
+## Running as a daemon on Linux through `systemd`
+
+In this section we show how to run the ConCom agent as a daemon on Linux through
+systemd. First let's make sure that your Linux distribution uses systemd. You
+can check this by running the following command:
+
+```bash
+ls -lh /sbin/init
+```
+If you have a `systemd` based system it should show a symlink to `systemd`
+something like the following:
+```
+lrwxrwxrwx. 1 root root 22 Oct 15 14:17 /sbin/init -> ../lib/systemd/systemd
+```
+If you don't have systemd you can check out the section `Running as a daemon on
+non-systemd based Linux` below.
+
+### Step 1: Download the ConCom agent
+
+Download the ConCom agent for your architecture from the [releases] page
+[here](https://github.com/https://github.com/compliance-framework/agent/releases)
+and place it in a directory of your choice. For example, you can download the
+agent for Linux x86_64 as follows:
+
+```bash
+CONCOM_RELEASE=0.1.0
+ARCH=x86_64
+OS=Linux
+curl -LOf https://github.com/compliance-framework/agent/releases/download/v${CONCOM_RELEASE}/agent_${OS}_${ARCH}.tar.gz
+```
+
+Then you need to extract the agent and copy it to a directory of your choice.
+For example you can run the following commands:
+
+```bash
+tar xvf agent_${OS}_${ARCH}.tar.gz
+sudo cp agent /usr/local/bin/concom-agent
+```
+
+### Step 2: Create a systemd service file
+
+Create a systemd service file for the ConCom agent. You can use the following:
+
+```bash
+sudo tee /etc/systemd/system/concom-agent.service <<EOF
+[Unit]
+Description=Continuous Compliance (ConCom) Agent
+Documentation=https://github.com/continuouscompliance/agent
+After=network.target local-fs.target
```

Comment:
```suggestion
Wants=network-online.target
After=network.target network-online.target
```

`network-online` for completeness?

Reply:
I stole this from `containerd` I think, or something like that, but sure, can't hurt. I suppose our use case is it's pointless doing this if the machine is offline and can't send results out where you could still run `containerd` on a laptop without networking a wifi connection say. I'll commit it.